src/pages/shopping-cart.vue

```diff
@@ -0,0 +1,263 @@
+
+import walletMixin from '~/mixins/wallet';
+import walletLoginMixin from '~/mixins/wallet-login';
+
+export default {
+  name: 'ShoppingCartPage',
+  filters: {
+    formatNumberWithLIKE,
+  },
+  mixins: [walletMixin, walletLoginMixin],
+  computed: {
+    ...mapGetters([
+      'getNFTClassPurchaseInfoById',
+      'getNFTClassListingInfoById',
+      'shoppingCartNFTClassList',
+    ]),
+    classIdList() {
+      return this.shoppingCartNFTClassList.map(item => item.classId);
+    },
+    totalNFTPrice() {
+      return this.shoppingCartNFTClassList.reduce((totalPrice, item) => {
+        const purchaseInfo = this.getNFTClassPurchaseInfoById(item.classId);
+        return totalPrice + purchaseInfo?.price * item.quantity;
+      }, 0);
+    },
+    grantAmount() {
+      return this.shoppingCartNFTClassList.reduce((totalPrice, item) => {
+        const purchaseInfo = this.getNFTClassPurchaseInfoById(item.classId);
+        return totalPrice + purchaseInfo?.totalPrice * item.quantity;
+      }, 0);
+    },
+    isInsufficientAmount() {
+      return this.walletLIKEBalance < this.grantAmount;
+    },
+  },
+  methods: {
+    ...mapActions([
+      'clearShoppingCart',
+      'removeNFTClassFromShoppingCart',
+      'walletFetchLIKEBalance',
+      'uiSetTxError',
+      'uiSetTxStatus',
+      'uiToggleCollectModal',
+    ]),
+    handleClickRemoveButton(classId) {
+      logTrackerEvent(
+        this,
+        'NFT',
+        'shopping_cart_remove_item',
+        this.getAddress,
+        1
+      );
+      this.removeNFTClassFromShoppingCart({ classId });
+    },
+    async handleClickCheckoutButton() {
+      logTrackerEvent(
+        this,
+        'NFT',
+        'shopping_cart_click_checkout_button',
+        this.getAddress,
+        1
+      );
+      try {
+        if (!this.getAddress || !this.getSigner) {
+          const isConnected = await this.connectWallet({
+            shouldSkipLogin: true,
+          });
+          if (!isConnected) return;
+        }
+
+        await this.walletFetchLIKEBalance();
+        if (this.isInsufficientAmount) {
+          this.uiToggleCollectModal({
+            classId: this.classIdList,
+            status: TX_STATUS.INSUFFICIENT,
+          });
+          this.uiSetTxError('INSUFFICIENT_BALANCE');
+          return;
+        }
+
+        this.uiToggleCollectModal({
+          classId: this.classIdList,
+          status: TX_STATUS.SIGN,
+        });
+
+        const signData = await signGrant({
+          senderAddress: this.getAddress,
+          amountInLIKE: this.grantAmount,
+          signer: this.getSigner,
+        });
+
+        this.uiSetTxStatus(TX_STATUS.PROCESSING);
+
+        const { txHash, code } = await broadcastTx(signData, this.getSigner);
+        if (code !== 0) {
+          throw new Error(`TX_FAILED_WITH_CODE_${code}`);
+        }
+        if (!txHash) {
+          throw new Error('TX_FAILED_WITHOUT_TX_HASH');
+        }
+
```

Comment:
Please set tx status processing non blocking

Reply:
Purchase should be blocking?

Reply:
The part we must block to prevent sequence/grant error is the authz grant tx, after the tx is completed and api received the request with tx hash, the remaining flow is automated